crates/goose-mcp/src/computercontroller/platform/mod.rs

```diff
@@ -0,0 +1,29 @@
+mod macos;
+mod windows;
+
+#[cfg(target_os = "windows")]
+pub use self::windows::WindowsAutomation;
+
+#[cfg(target_os = "macos")]
+pub use self::macos::MacOSAutomation;
+
+pub trait SystemAutomation: Send + Sync {
+    fn execute_system_script(&self, script: &str) -> std::io::Result<String>;
+    fn get_shell_command(&self) -> (&'static str, &'static str); // (shell, arg)
+    fn get_temp_path(&self) -> std::path::PathBuf;
+}
+
+pub fn create_system_automation() -> Box<dyn SystemAutomation + Send + Sync> {
```

Comment:
realizing that this extension prob doesn't work with linux machines? a follow up to this pr is to make that more compatible

Reply:
it does. at least partially thorugh powershell